exercises/robot-simulator/src/lib.rs

```diff
@@ -0,0 +1,44 @@
+// The code below is a stub. Just enough to satisfy the compiler.
+// In order to pass the tests you can add-to or change any of this code.
+
+#[derive(PartialEq, Debug)]
+pub enum Direction {
+    North,
+    East,
+    South,
+    West,
+}
+
+pub struct Robot;
+
+impl Robot {
+    #[allow(unused_variables)]
+    pub fn new(x: isize, y: isize, d: &Direction) -> Self {
```

Comment:
The `Direction` argument being a reference here is a bit out of place when compared to the rest of the interface, for if we assume the user may want to store `&Direction` in their `Robot` type, the other methods will not allow this to be updated to anything reasonable, i.e.:
```rust
fn turn_right(&self) -> Self {
     Robot { 
         position: self.position,
         // nowhere to tie this lifetime to as it's not an input
         direction: &self.direction.next_clockwise() 
     }
}
```